model/task_lifecycle.go

```diff
@@ -228,9 +236,19 @@ func TryResetTask(taskId, user, origin string, detail *apimodels.TaskEndDetail)
 		}
 	}
 
-	if err = resetTask(t.Id, user); err != nil {
+	if t.IsPartOfSingleHostTaskGroup() {
+		var tasks []task.Task
+		tasks, err = GetTasksInTaskGroup(t)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't get tasks in group '%s'", t.TaskGroup)
+		}
+		if err = resetManyTasks(tasks, user); err != nil {
+			return errors.WithStack(err)
+		}
+	} else if err = resetTask(t.Id, user); err != nil {
```

Comment:
why are resetManyTasks and restTasks not methods on a task?

Reply:
resetTask has been here for a while, but I believe the things it includes couldn't be in the task package (it involves updating builds and versions as well).

Reply:
alas.